sdk/src/process_instruction.rs

```diff
@@ -41,6 +47,10 @@ pub trait InvokeContext {
     ) -> Result<(), InstructionError>;
     /// Get the program ID of the currently executing program
     fn get_caller(&self) -> Result<&Pubkey, InstructionError>;
+    /// Removes the first keyed account
+    fn pop_first_keyed_account(&mut self);
```

Comment:
"pop" is usually used for the constant-time operation to pop an item off the top of a stack.  Rust libs usually use "next" to remove the first item from an iterator. Should we call this "next_keyed_account"?

Reply:
This actually pops the front (of a queue if you will), and it is constant time as it operates on a ref of a slice.

Reply:
But, you are right in that it is similar to the `next` method of iterators.
However, it only advances the internal pointers and does not return anything.

Reply:
`remove_first`?